libcudacxx/include/cuda/std/detail/libcxx/include/optional

```diff
@@ -264,6 +264,7 @@ struct __optional_destruct_base<_Tp, false>
     };
     bool __engaged_;
 
+    _CCCL_EXEC_CHECK_DISABLE
     _LIBCUDACXX_INLINE_VISIBILITY
     _CCCL_CONSTEXPR_CXX20 ~__optional_destruct_base()
```

Comment:
I need this to allow `cuda::std::optional` to destruct an `nvtx3::scoped_range_in`. @miscco is aware and may provide a more generic PR in-time.

Reply:
I am fine with you adding this as a workaround, but please also add it to the other specialization of `optional_destruct_base`

Reply:
I am still counting on your PR #1663 to land faster, but if not, I will add the annotations here.

Reply:
The other specialization of `optional_destruct_base` is for trivially destructible types and hence has no dtor :)